src/test/java/com/kaziamyr/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,313 @@
+            scripts = "classpath:database/categories/add-fiction-to-categories-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+    )
+    @Sql(
+            scripts = "classpath:database/"
+                    + "delete-all-from-categories-books-books_categories-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Test create() with a correct book")
+    void create_withValidBook_returnBookDto() throws Exception {
+        CreateBookRequestDto zakharBerkutRequest = new CreateBookRequestDto()
+                .setTitle("Zakhar Berkut")
+                .setAuthor("Ivan Franko")
+                .setIsbn("9788339123867")
+                .setPrice(BigDecimal.valueOf(10))
+                .setDescription("description")
+                .setCoverImage("https://example.com/updated-cover-image.jpg")
+                .setCategoryIds(new Long[]{1L});
+        MvcResult mvcResult = mockMvc.perform(post("/books")
+                        .content(objectMapper.writeValueAsString(zakharBerkutRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actual =
+                objectMapper.readValue(mvcResult.getResponse().getContentAsString(), BookDto.class);
+
+        assertTrue(EqualsBuilder.reflectionEquals(ZAKHAR_BERKUT_DTO, actual, "id"));
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @Sql(
+            scripts = "classpath:database/categories/add-fiction-to-categories-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+    )
+    @Sql(
+            scripts = "classpath:database/"
+                    + "delete-all-from-categories-books-books_categories-tables.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Test create() with an invalid book")
+    void create_withInvalidBook_throwException() throws Exception {
+        CreateBookRequestDto invalidZakharBerkutRequest = new CreateBookRequestDto()
+                .setTitle("Zakhar Berkut")
+                .setAuthor("Ivan Franko")
+                .setIsbn("97883391238674")
+                .setPrice(BigDecimal.valueOf(10))
+                .setDescription("description")
+                .setCoverImage("https://example.com/updated-cover-image.jpg")
+                .setCategoryIds(new Long[]{1L});
+        mockMvc.perform(post("/books")
+                        .content(objectMapper.writeValueAsString(invalidZakharBerkutRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isBadRequest());
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @Sql(
+            scripts = {
+                    "classpath:database/categories/add-fiction-to-categories-table.sql",
```

Comment:
Maybe put these strings with scripts destination in some class with constants to have less text?